Shopify/dawn
# [Discounts] fix: show accurate subtotal price for cart

### PR Summary: 

Updates total price to be items_subtotal_price to more accurately represent the value after line item discounts only.

Before:

<img width="1173" alt="image" src="https://github.com/Shopify/dawn/assets/88739497/1469287c-a6e7-4d54-af14-7cc95dd4109b">

After:

![image](https://github.com/Shopify/dawn/assets/88739497/4f155c53-e7cf-43ef-86d8-2787b2a83ef8)


![image](https://github.com/Shopify/dawn/assets/88739497/b508808d-c518-44f8-82d0-021e212a072e)


### Why are these changes introduced?

fixes: https://github.com/Shopify/core-issues/issues/57440
fixes: https://github.com/Shopify/dawn/issues/2781


### What approach did you take?

### Other considerations

### Decision log

| # | Decision | Alternatives | Rationale | Downsides |
|---|---|---|---|---|
| 1 |   |   |   |   |


### Visual impact on existing themes



### Testing steps/scenarios

manually edited in store editor: https://admin.web.fix-inaccurate-total.mathius-johnson.us.spin.dev/store/shop1/themes/1?key=sections%2Fmain-cart-footer.liquid

cart: https://shop1.shopify.fix-inaccurate-total.mathius-johnson.us.spin.dev/cart

- add products to cart
- go to checkout and add product + order discounts (take note of total price after order discount)
- go back to cart and confirm subtotal accurately reflects amount of line items - line item discounts, order discount not included (should be a higher amount than total price with order discount applied)

### Demo links


- [Store](https://shop1.shopify.fix-inaccurate-total.mathius-johnson.us.spin.dev/cart)
- [Editor](https://admin.web.fix-inaccurate-total.mathius-johnson.us.spin.dev/store/shop1/themes/1?key=sections%2Fmain-cart-footer.liquid)

### Checklist
